src/docs/arch-s3-extension.dox

```diff
@@ -0,0 +1,78 @@
+/*! @arch_page arch-s3-extension S3 Storage Source
+
+@section s3_extension_storage_source S3 Storage Source Implementation
+
+S3 storage source, or the S3 store, is a WiredTiger extension that implements the storage source
+abstraction for AWS's S3 object-store. The store extends the WiredTiger's capability to store the
+data files on the S3 cloud. The extension provides a means to construct, configure and control a
+custom file system linked to an S3 bucket. The extension internally uses the API provided by the
+AWS C++ SDK to interact with the S3 cloud.
+
+The S3 store implements the following key functionalities defined by the storage source interface:
+
+- Initialize the store, the AWS SDK and create a logging module for the store and the SDK.
+
+- Register the S3 store as a storage source available to Wiredtiger.
+
+- Provide an API to access and terminate the store. A reference count is incremented each time the
+storage source reference is taken and decremented with the termination. When the reference count
+goes to zero, the storage source is de-registered with WiredTiger and any allocated resources freed.
+
+- Another essential functionality the S3 store provides is to create a \c WT_FILE_SYSTEM
+implementation for a given bucket. The store requires a bucket and an access token to create the
+filesystem. The bucket is expected to be a string argument, which is the bucket name and the
+corresponding region separated by a ';', e.g. "bucket1;ap-southeast-2". The AWS requires an access
+token to authenticate programmatic access to S3. The access token comprises an access key ID and a
+secret access key as a set. The S3 store requires the token to be provided as a single string
+argument of the access key followed by the secret key, separated by a ';'.
+
+- The S3 store also configures a directory on the file system local to the WiredTiger database as a
+cache for the object files. The cache is meant to improve the read latency for the object files
```

Comment:
Can we use either "The cache is used to improve" or "The cache improves" unless it is uncertain whether it improves it or not?